app/content-panel/blogs/[id]/page.tsx

```diff
@@ -0,0 +1,113 @@
+'use client';
+
+import React, { useState, useEffect } from 'react';
+import styles from './BlogDetailsPage.module.scss';
+import BlogDetailsCard from '../../../../components/reusable-components/blogDetails-card/BlogDetailsCard';
+
+interface Author {
+  first_name: string;
+  last_name: string;
+}
+
+interface BlogDetails {
+  title: string;
+  content: string;
+  image: string;
+  author: Author;
+  publishDate: string;
+  tags: string[];
+}
+
+const BlogDetailsPage = ({ params }: { params: { id: string } }) => {
+  const [isEditable, setIsEditable] = useState(false);
+  const [selectedImage, setSelectedImage] = useState<string | null>(null);
+  const [blogDetails, setBlogDetails] = useState<BlogDetails>({
+    title: 'N/A',
+    content: 'N/A',
+    image: '',
+    author: { first_name: 'N/A', last_name: 'N/A' },
+    publishDate: 'N/A',
+    tags: [],
+  });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const response = await fetch(
+        `${process.env.NEXT_PUBLIC_API_BASE_URL}/blog-posts/${params.id}`
+      );
+      const data = await response.json();
+      const { title, content, image, author, publish_date: publishDate, tags } = data.data || {};
+
+      setBlogDetails({
+        title: title || 'N/A',
+        content: content || 'N/A',
+        image: image || '',
+        author: author || { first_name: 'N/A', last_name: 'N/A' },
+        publishDate: publishDate || 'N/A',
+        tags: tags || [],
+      });
+    };
+    fetchData();
+  }, [params.id]);
+
+  const handleEditClick = () => {
```

Comment:
Just a tip when you work on this ticket next, please investigate if memoization is needed.

Reply:
I ll do that